Assets/Scripts/Grid/Positioning/SpiralSequenceRandomPositionProvider.cs

```diff
@@ -0,0 +1,109 @@
+                                                    IRandomProvider randomProvider){
+            _randomProvider = randomProvider;
+            _gridPositionCalculator = gridPositionCalculator;
+        }
+        
+        public Vector2[] GetRandomUniquePositions(IGrid grid, int maxDistanceFromCenter, int numTilesToGenerate){
+            Vector2 spiralCenter = _gridPositionCalculator.GetTileClosestToCenter(grid);
+            Vector2 spiralEndTile = new Vector2(spiralCenter.x - numTilesToGenerate, 
+                                                spiralCenter.y - numTilesToGenerate);
+            LinkedList<Vector2> availableTiles = new LinkedList<Vector2>();
+            
+            int k = 1;
+            Vector2 currentPosition, stretchStart = spiralCenter;
+
+            do {
+                currentPosition = new Vector2(stretchStart.x, stretchStart.y);
+                for (int i = (int) stretchStart.y; i > stretchStart.y - k; i--) {
+                    currentPosition.y = i;
+                    AddIfAvailableTile(availableTiles, grid, currentPosition);
+                }
+                
+                //If at this point the position is (center.x - maxDistance, center.y - maxDistance) the spiral is
+                //complete for maxDistance.
+                if (currentPosition == spiralEndTile) {
+                    break;
+                }
+
+                stretchStart = new Vector2(currentPosition.x, currentPosition.y - 1);
+
+                currentPosition = new Vector2(stretchStart.x, stretchStart.y);
+                for (int i = (int) stretchStart.x; i < stretchStart.x + k; i++) {
+                    currentPosition.x = i;
+                    AddIfAvailableTile(availableTiles, grid, currentPosition);
+                }
+
+                stretchStart = new Vector2(currentPosition.x + 1, currentPosition.y);
+                k++;
+
+                currentPosition = new Vector2(stretchStart.x, stretchStart.y);
+                for (int i = (int) stretchStart.y; i < stretchStart.y + k; i++) {
+                    currentPosition.y = i;
+                    AddIfAvailableTile(availableTiles, grid, currentPosition);
+                }
+
+                stretchStart = new Vector2(currentPosition.x, currentPosition.y + 1);
+
+                currentPosition = new Vector2(stretchStart.x, stretchStart.y);
+                for (int i = (int) stretchStart.x; i > stretchStart.x - k; i--) {
+                    currentPosition.x = i;
+                    AddIfAvailableTile(availableTiles, grid, currentPosition);
+                }
+
+                stretchStart = new Vector2(currentPosition.x - 1, currentPosition.y);
+                k++;
+            } while (true);
+
+            return ChooseRandomPositionsFromAvailableTiles(availableTiles, numTilesToGenerate);
+        }
+
+        private void AddIfAvailableTile(LinkedList<Vector2> availableTiles, IGrid grid, Vector2 tile) {
```

Comment:
A better name for this method would be `AddTileIfInsideGridBounds`

Reply:
I agree.